### PR TITLE
[CP] Send email to committee request support when CP is approve

### PR DESCRIPTION
**Pull request by NyKo24**

### CODE REVIEW

File: src/Controller/EnMarche/CitizenProjectController.php [hunk 1]
```diff
@@ -77,4 +77,51 @@ public function committeeAutocompleteAction(Request $request)
 
         return new JsonResponse($result ?? []);
     }
+
+    /**
+     * @Route("/mon-comite-soutien/{slug}", name="app_citizen_project_committee_support")
+     * @Method("GET")
```
Comment (B-Galati) [inline src/Controller/EnMarche/CitizenProjectController.php]: This action should be `POST`. 
There is also a security issue and a CSRF token should be used to fix it.

File: src/Controller/EnMarche/CitizenProjectController.php [hunk 2]
```diff
@@ -77,4 +77,51 @@ public function committeeAutocompleteAction(Request $request)
...
+    public function committeeSupportAction(CitizenProject $citizenProject): Response
+    {
+        if (!$this->getUser()->isSupervisor()) {
+            throw $this->createAccessDeniedException();
+        }
+
+        $committeeManager = $this->get('app.committee.manager');
+        $flashMessage = sprintf('Le projet citoyen %s n\'a pas demandé votre soutient', $citizenProject->getName());
+
+        foreach ($citizenProject->getCommitteeSupports() as $committeeSupport) {
```
Comment (B-Galati) [inline src/Controller/EnMarche/CitizenProjectController.php]: Too much work performed in this controller. Try to move the business logic to the entity itself and/or manager/voter depending on the context.

File: templates/citizen_project/_sidebar.html.twig [hunk 3]
```diff
@@ -64,14 +64,14 @@
...
-                    </li>
-                {% endif %}
+            {% for committeeSupport in citizen_project.approvedCommitteeSupports %}
+                <li class="text--body">
+                    {{ committeeSupport.committee }}
+                </li>
             {% endfor %}
         </ul>
-        <a href="#">Notre comité souhaite aussi soutenir ce projet</a>
+        {% if app.user and is_granted('ROLE_SUPERVISOR') %}
```
Comment (B-Galati) [inline templates/citizen_project/_sidebar.html.twig]: I think testing `if app.user` is implicit when using `is_granted`

File: src/Controller/EnMarche/CitizenProjectController.php (hunk 1 above)
Comment (NyKo24) [reply src/Controller/EnMarche/CitizenProjectController.php]: this URL is call on email send to committee supervisor

File: templates/citizen_project/_sidebar.html.twig (hunk 3 above)
Comment (NyKo24) [reply templates/citizen_project/_sidebar.html.twig]: 👍

File: src/Controller/EnMarche/CitizenProjectController.php (hunk 1 above)
Comment (B-Galati) [reply src/Controller/EnMarche/CitizenProjectController.php]: Ok but we still need to fix the security issue.
I see 2 solutions : 
    - Create an intermediary page that would ask a confirmation to the user with a form (the simplest and best solution to me).
    - Use any kind of token (JWT, hmac time limited, a one time random secret, or whatever). Just like we do with password reset request.

File: src/CitizenProject/CitizenProjectMessageNotifier.php [hunk 4]
```diff
@@ -65,6 +72,19 @@ private function scheduleCreationNotification(CitizenProject $citizenProject): v
         ]));
     }
 
+    private function sendAskCommitteeSupport(CitizenProject $citizenProject, Adherent $creator): void
+    {
+        foreach ($citizenProject->getPendingCommitteeSupports() as $committeeSupport) {
+            $this->mailer->sendMessage(
+                CitizenProjectRequestCommitteeSupportMessage::create(
+                    $citizenProject,
+                    $creator,
```
Comment (HeahDude) [inline src/CitizenProject/CitizenProjectMessageNotifier.php]: Passing the creator all along is not needed, just call `$citizenProject->getCreator()` in that `create` method.

File: src/Controller/EnMarche/CitizenProjectController.php (hunk 1 above)
Comment (HeahDude) [reply src/Controller/EnMarche/CitizenProjectController.php]: 👍 to send a temporary token, as done with account creation confirmation.

File: src/Entity/CitizenProjectCommitteeSupport.php [hunk 5]
```diff
@@ -88,7 +88,7 @@ public function isPending(): bool
         return self::PENDING === $this->status;
     }
 
-    public function isApprove(): bool
+    public function isApproved(): bool
     {
         return self::APPROVE === $this->status;
```
Comment (illusionOfParadise) [inline src/Entity/CitizenProjectCommitteeSupport.php]: It's better to name a constant as `APPROVED`

File: src/Entity/CitizenProjectCommitteeSupport.php [hunk 6]
```diff
@@ -98,4 +98,9 @@ public function approve(string $timestamp = 'now'): void
         $this->status = self::APPROVE;
         $this->approvedAt = new \DateTime($timestamp);
     }
+
+    public function getCitizenProject()
```
Comment (illusionOfParadise) [inline src/Entity/CitizenProjectCommitteeSupport.php]: Missing type hint

**Pull request merged**